### PR TITLE
Update ZAP schema & add ThreeLevelAutoEnum

### PR DESCRIPTION
**Pull request by gmarcosb**
Pre-step for https://github.com/project-chip/connectedhomeip/pull/35773:

1) Add `ThreeLevelAutoEnum`, added @ https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/10004
2) Add missing schema definitions to `zcl.xsd`

ZAP files generated using:
```
 ./scripts/tools/zap_regen_all.py
```

### CODE REVIEW

File: src/app/zap-templates/zcl/zcl.xsd [hunk 1]
```diff
@@ -8,6 +8,11 @@ This schema describes the format of the XML files, that describe the ZCL specifi
 (c) Copyright 2012, 2021, Silicon Laboratories Inc.
   </xs:documentation>
   </xs:annotation>
+  <xs:simpleType name="integerWithHex">
+    <xs:restriction base="xs:string">
+      <xs:pattern value="[-+]?(0x[0-9A-Fa-f]+|[0-9]+)"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/zcl.xsd:13]: Are there any uses of `-` followed by hex?

File: src/app/zap-templates/zcl/zcl.xsd [hunk 2]
```diff
@@ -18,14 +23,20 @@ This schema describes the format of the XML files, that describe the ZCL specifi
...
+    <xs:attribute name="optional" type="xs:boolean" default="false"/>
+    <xs:attribute name="min" type="integerWithHex"/>
+    <xs:attribute name="max" type="integerWithHex"/>
   </xs:complexType>
   <xs:complexType name="tag">
     <xs:attribute name="name" use="required" type="xs:string"/>
     <xs:attribute name="description" use="required" type="xs:string"/>
   </xs:complexType>
   <xs:complexType name="arg">
+    <xs:attribute name="id" type="integerWithHex" />
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/zcl.xsd:37]: ids can't be negative... also, above eventField uses string for id?

File: src/app/zap-templates/zcl/zcl.xsd [hunk 3]
```diff
@@ -285,10 +317,28 @@ This schema describes the format of the XML files, that describe the ZCL specifi
...
+              </xs:complexType>
+            </xs:element>
+          </xs:sequence>
+          </xs:complexType>
+        </xs:element>
+      </xs:choice>
       <xs:attribute name="introducedIn" type="xs:string"/>
       <xs:attribute name="manufacturerCode" type="zclCode"/>
       <xs:attribute name="singleton" type="xs:boolean"/>
+      <xs:attribute name="apiMaturity" type = "zclApiMaturity" />
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/zcl.xsd:341]: Why the extra spaces around `=`?

File: src/app/zap-templates/zcl/zcl.xsd [hunk 4]
```diff
@@ -342,8 +392,9 @@ This schema describes the format of the XML files, that describe the ZCL specifi
         <xs:element minOccurs="0" maxOccurs="unbounded" name="cluster" type="clusterRef"/>
         <xs:element minOccurs="0" maxOccurs="unbounded" name="item" type="item"/>
       </xs:sequence>
-      <xs:attribute name="length" type="xs:integer"/>
+      <xs:attribute name="length" type="integerWithHex"/>
       <xs:attribute name="name" use="required" type="xs:string"/>
+      <xs:attribute name="apiMaturity" type = "zclApiMaturity" />
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/zcl.xsd:397]: Why the extra spaces around `=`?

**Review by andy31415 — APPROVED** [review]

**Pull request merged**